### ISSUE TITLE
[Bug] InvalidCastException when setting `border-radius` in CSS

### ISSUE THREAD

**Issue by chkn** [xamarin/Xamarin.Forms#6361]
### Description

There is an `InvalidCastException` thrown when setting `border-radius` in CSS. It doesn't always occur, but I have been able to reproduce it with a Button that causes a Frame to be added to the layout (see attached test case)

### Steps to Reproduce

1. Download and run the attached test case
2. Click the button
3. Note the exception

Note that I've only tested with the iOS head, not sure if this also affects Android.

**UPDATE:** Sam has confirmed that it affects Android as well.

### Expected Behavior

The frame should be added with no incident.

### Actual Behavior

This exception is thrown, which crashes the app:

```
System.InvalidCastException: Specified cast is not valid.
  at Xamarin.Forms.Button.CornerRadiusPropertyChanged (Xamarin.Forms.BindableObject bindable, System.Object oldvalue, System.Object newvalue) [0x00005] in D:\a\1\s\Xamarin.Forms.Core\Button.cs:298 
  at Xamarin.Forms.BindableObject.SetValueActual (Xamarin.Forms.BindableProperty property, Xamarin.Forms.BindableObject+BindablePropertyContext context, System.Object value, System.Boolean currentlyApplying, Xamarin.Forms.Internals.SetValueFlags attributes, System.Boolean silent) [0x00120] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:445 
  at Xamarin.Forms.BindableObject.SetValueCore (Xamarin.Forms.BindableProperty property, System.Object value, Xamarin.Forms.Internals.SetValueFlags attributes, Xamarin.Forms.BindableObject+SetValuePrivateFlags privateAttributes) [0x00173] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:379 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle, System.Boolean checkAccess) [0x00042] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:316 
  at Xamarin.Forms.BindableObject.SetValue (Xamarin.Forms.BindableProperty property, System.Object value, System.Boolean fromStyle) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\BindableObject.cs:303 
  at Xamarin.Forms.StyleSheets.Style.Apply (Xamarin.Forms.VisualElement styleable, System.Boolean inheriting) [0x0007f] in D:\a\1\s\Xamarin.Forms.Core\StyleSheets\Style.cs:73 
  at Xamarin.Forms.StyleSheets.StyleSheet.ApplyCore (Xamarin.Forms.Element styleable) [0x00037] in D:\a\1\s\Xamarin.Forms.Core\StyleSheets\StyleSheet.cs:114 
  at Xamarin.Forms.StyleSheets.StyleSheet.Apply (Xamarin.Forms.Element styleable) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\StyleSheets\StyleSheet.cs:99 
  at Xamarin.Forms.StyleSheets.StyleSheet.Xamarin.Forms.IStyle.Apply (Xamarin.Forms.BindableObject bindable) [0x0000b] in D:\a\1\s\Xamarin.Forms.Core\StyleSheets\StyleSheet.cs:94 
  at Xamarin.Forms.Element.ApplyStyleSheetsOnParentSet () [0x00045] in D:\a\1\s\Xamarin.Forms.Core\Element_StyleSheets.cs:55 
  at Xamarin.Forms.Element.OnParentSet () [0x00017] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:347 
  at Xamarin.Forms.NavigableElement.OnParentSet () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Shell\NavigableElement.cs:55 
  at Xamarin.Forms.Element.set_Parent (Xamarin.Forms.Element value) [0x0008e] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:208 
  at Xamarin.Forms.Element.OnChildAdded (Xamarin.Forms.Element child) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Element.cs:322 
  at Xamarin.Forms.VisualElement.OnChildAdded (Xamarin.Forms.Element child) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\VisualElement.cs:742 
  at Xamarin.Forms.Layout`1[T].OnChildAdded (Xamarin.Forms.Element child) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:32 
  at Xamarin.Forms.Layout.OnInternalAdded (Xamarin.Forms.View view) [0x0001d] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:453 
  at Xamarin.Forms.Layout.InternalChildrenOnCollectionChanged (System.Object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00078] in D:\a\1\s\Xamarin.Forms.Core\Layout.cs:443 
  at (wrapper delegate-invoke) <Module>.invoke_void_object_NotifyCollectionChangedEventArgs(object,System.Collections.Specialized.NotifyCollectionChangedEventArgs)
  at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedEventArgs e) [0x00018] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/external/corefx/src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs:263 
  at System.Collections.ObjectModel.ObservableCollection`1[T].OnCollectionChanged (System.Collections.Specialized.NotifyCollectionChangedAction action, System.Object item, System.Int32 index) [0x00000] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/external/corefx/src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs:338 
  at System.Collections.ObjectModel.ObservableCollection`1[T].InsertItem (System.Int32 index, T item) [0x0001a] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/external/corefx/src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs:196 
  at System.Collections.ObjectModel.Collection`1[T].Add (T item) [0x00020] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/external/corefx/src/Common/src/CoreLib/System/Collections/ObjectModel/Collection.cs:73 
  at Xamarin.Forms.ObservableWrapper`2[TTrack,TRestrict].Add (TRestrict item) [0x0004b] in D:\a\1\s\Xamarin.Forms.Core\ObservableWrapper.cs:35 
  at BorderRadiusRepro.MainPage.Handle_Clicked (System.Object sender, System.EventArgs e) [0x00001] in /Users/alex/Projects/BorderRadiusRepro/BorderRadiusRepro/MainPage.xaml.cs:21 
  at Xamarin.Forms.Button.Xamarin.Forms.Internals.IButtonElement.PropagateUpClicked () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Button.cs:177 
  at Xamarin.Forms.ButtonElement.ElementClicked (Xamarin.Forms.VisualElement visualElement, Xamarin.Forms.Internals.IButtonElement ButtonElementManager) [0x0001f] in D:\a\1\s\Xamarin.Forms.Core\ButtonElement.cs:61 
  at Xamarin.Forms.Button.SendClicked () [0x00000] in D:\a\1\s\Xamarin.Forms.Core\Button.cs:163 
  at Xamarin.Forms.Platform.iOS.ButtonElementManager.OnButtonTouchUpInside (Xamarin.Forms.IButtonController element) [0x0000c] in <db147fad2aa9412c845ff0ead92dfe20>:0 
  at Xamarin.Forms.Platform.iOS.ButtonRenderer.OnButtonTouchUpInside (System.Object sender, System.EventArgs eventArgs) [0x00006] in <db147fad2aa9412c845ff0ead92dfe20>:0 
  at UIKit.UIControlEventProxy.Activated () [0x00004] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/UIKit/UIControl.cs:38 
--- End of stack trace from previous location where exception was thrown ---

  at (wrapper managed-to-native) UIKit.UIApplication.UIApplicationMain(int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) [0x00005] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:79 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0002c] in /Library/Frameworks/Xamarin.iOS.framework/Versions/12.8.0.2/src/Xamarin.iOS/UIKit/UIApplication.cs:63 
  at BorderRadiusRepro.iOS.Application.Main (System.String[] args) [0x00001] in /Users/alex/Projects/BorderRadiusRepro/BorderRadiusRepro.iOS/Main.cs:15
```

### Basic Information

- Version with issue: 3.6, 4.0
- Last known good version: 3.4
- IDE: VS Mac
- Platform Target Frameworks: 
  - iOS:  Xcode 10.2.1 (14490.122)
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

Xamarin.iOS
Version: 12.8.0.2 (Visual Studio Professional)
Hash: f2248ae6
Branch: d16-0
Build date: 2019-04-23 11:59:04-0400

Xamarin.Android
Version: 9.2.3.0 (Visual Studio Professional)
Android SDK: /Users/alex/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		4.4 (API level 19)
		6.0 (API level 23)
		7.1 (API level 25)
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.0
SDK Build Tools Version: 28.0.0

### Reproduction Link

[BorderRadiusRepro.zip](https://github.com/xamarin/Xamarin.Forms/files/3237603/BorderRadiusRepro.zip)

**Comment by samhouts** [comment]
Confirmed it also affects Android.

**Comment by StephaneDelcroix** [comment]
I can repro in a unit test

**Comment by samhouts** [comment]
closed #6401
